src/loki/trace_route_action.cc

```diff
@@ -40,14 +40,31 @@ void check_shape(const google::protobuf::RepeatedPtrField<valhalla::Location>& s
 
 void check_distance(const google::protobuf::RepeatedPtrField<valhalla::Location>& shape,
                     float max_distance,
+                    float max_breakage_distance,
                     float max_factor = 1.0f) {
+
   // Adjust max - this enables max edge_walk distance to be larger
   max_distance *= max_factor;
 
-  // Calculate "crow distance" of shape
-  auto crow_distance = to_ll(*shape.begin()).Distance(to_ll(*shape.rbegin()));
+  bool can_be_matched = false;
+  float crow_distance = 0.f;
+  for (auto iter = shape.begin(); iter < shape.end() - 1; ++iter) {
+    // We bail when the distance between two locations exceeds the threshold
+    PointLL curr_point = to_ll(*iter);
+    PointLL next_point = to_ll(*std::next(iter));
+    float point_to_point_distance = curr_point.Distance(next_point);
+    crow_distance += point_to_point_distance;
+
+    if (point_to_point_distance <= max_breakage_distance) {
+      can_be_matched = true;
+    }
+
+    if (crow_distance > max_distance) {
+      throw valhalla_exception_t{154};
+    }
+  }
 
-  if (crow_distance > max_distance) {
+  if (!can_be_matched) {
     throw valhalla_exception_t{154};
```

Comment:
this returns a custom message that is not native to OSRM. see https://github.com/valhalla/valhalla/blob/master/src/worker.cc#L190

we should check what OSRM does when you give it only points that are way too far away to be matched.

Reply:
yeah i think this one should be a new error code as we discussed offline.